spring-integration-redis/src/main/java/org/springframework/integration/redis/store/RedisMessageStore.java

```diff
@@ -69,12 +82,27 @@ public RedisMessageStore(RedisConnectionFactory connectionFactory) {
 	 * @see AbstractKeyValueMessageStore#AbstractKeyValueMessageStore(String)
 	 */
 	public RedisMessageStore(RedisConnectionFactory connectionFactory, String prefix) {
+		this(connectionFactory, prefix, new DefaultLockRegistry());
+	}
+
+	/**
+	 * Construct {@link RedisMessageStore} based on the provided
+	 * {@link RedisConnectionFactory} and prefix.
+	 * @param connectionFactory the RedisConnectionFactory to use
+	 * @param prefix the key prefix to use, allowing the same broker to be used for
+	 * multiple stores.
+	 * @param lockRegistry the {@link LockRegistry} to use.
+	 * @since 6.4.1
+	 * @see AbstractKeyValueMessageStore#AbstractKeyValueMessageStore(String)
+	 */
+	public RedisMessageStore(RedisConnectionFactory connectionFactory, String prefix, LockRegistry lockRegistry) {
```

Comment:
I'm fine with `DefaultLockRegistry` as default, but let's make this as a setter! It is optional from object creation perspective, so no need in extra ctor argument!